### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Juakata**

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "../lib/player.rb"
+require "../lib/board.rb"
+
+play_again = true
+options = [1,2,3,4,5,6,7,8,9]
+
+while play_again
```
Comment (Ispirett) [inline bin/main.rb]: You can put all this logic in a class called game instead of having it in the bin class

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Board
+  def initialize
+    @wins1 = { 1 => [1, 2, 3], 2 => [1, 5, 9], 3 => [1, 4, 7], 4 => [2, 1, 3], 5 => [2, 5, 8], 6 => [3, 1, 2] }
```
Comment (Ispirett) [inline lib/board.rb]: The logic here is too hardcoded, a better approach is to make use of methods like `all?, any?, or 1,,3 (ranges)

File: lib/player.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Player
+  attr_reader :player
+  attr_accessor :winner
+  def initialize(player)
+    @player = player
+    @winner = false
```
Comment (Ispirett) [inline lib/player.rb]: All winner related logic should be handled by the board class

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require '../lib/player.rb'
```
Comment (bolah2009) [inline lib/game.rb]: #### `OPTIONAL` 

Let's use `require_relative` here instead. `require_relative` should be used when the file you need to load exists somewhere relative to the file that calls for the loading. Learn more [here](https://ruby-doc.org/core-2.1.2/Kernel.html#method-i-require_relative)

```suggestion
require_relative 'player.rb'
```

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+require '../lib/board.rb'
```
Comment (bolah2009) [inline lib/game.rb]: ### Consider using `require_relative` instead.

```suggestion
require_relative 'board.rb'
```

File: bin/main.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require '../lib/game.rb'
```
Comment (bolah2009) [inline bin/main.rb]: ### Consider using `require_relative` instead.

```suggestion
require_relative '../lib/game.rb'
```

File: lib/board.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      @map[i] = '.'
+      i += 1
+    end
+    @winner = 0
+  end
+
+  def display_board
+    i = 1
+    3.times do
+      print "\n #{@map[i]} | #{@map[i + 1]} | #{@map[i + 2]} \n"
```
Comment (bolah2009) [inline lib/board.rb]: #### `REQUIRED`

`main.rb` is the only place where `puts`, `print`, `p` and `get.chomp` can be used.  :blush: 
Let's refactor the code and do this... :+1: 

> Add executable file named `main.rb` to bin directory. This file should be responsible for user interaction. This is the only one place where `puts` and `get.chomp` can be used.

Let's do this for all other occurrences.

File: bin/main.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    end
+    while !board.add_map(move.to_i,player2.player.to_s)
+      print "\nOption already used, select a different option: "
+      move = gets.chomp
+    end
+    board.winner = 2 if board.check_map(move.to_i,player2.player.to_s)
+    break if board.winner != 0
+  end
+  board.display_board
+  if board.winner == 1
```
Comment (bolah2009) [inline bin/main.rb]: #### `REQUIRED`

## :bug: Are we testing for `draw`? :thinking: 

```ruby
$ ruby bin/main.rb

Select O or X for player 1: O

 . | . | . 
-----------
 . | . | . 
-----------
 . | . | . 

Player 1 options [1-9]: 7

 . | . | . 
-----------
 . | . | . 
-----------
 O | . | . 


---


Player 2 options [1-9]: 2

 O | X | . 
-----------
 X | X | O 
-----------
 O | O | X 

Player 1 options [1-9]: 3

 O | X | O 
-----------
 X | X | O 
-----------
 O | O | X 

Player 2 options [1-9]: 1

Option already used, select a different option: 9

Option already used, select a different option: DRAW
```

#### Currently, the game can't detect if it's a draw. :disappointed:

File: bin/main.rb [hunk 9]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require_relative '../lib/game.rb'
+game = Game.new()
+while game.play_again
+  game.play_again = false
+  print "\nSelect O or X for player 1: "
+  symbol = gets.chomp
+  symbol.capitalize!
+  while !game.players.include?symbol
```
Comment (Ispirett) [inline bin/main.rb]: This  code can be wrapped in a method.

File: bin/main.rb [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+  while true
+    print board.display_board
+    print "\nPlayer 1 options [1-9]: "
+    move = gets.chomp
+    while !game.options.include? move.to_i
+      print "\nPleas select from 1 to 9: "
+      move = gets.chomp
+    end
+    while !board.add_map(move.to_i,player1.player.to_s)
```
Comment (Ispirett) [inline bin/main.rb]: This code can be wrapped in a method.

File: bin/main.rb [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    break if board.winner != 0
+    break if board.plays == 9
+    print board.display_board
+    print "\nPlayer 2 options [1-9]: "
+    move = gets.chomp
+    while !game.options.include? move.to_i
+      print "\nPleas select from 1 to 9: "
+      move = gets.chomp
+    end
+    while !board.add_map(move.to_i,player2.player.to_s)
```
Comment (Ispirett) [inline bin/main.rb]: this code was already created and the code can be reused if it was within a method and add a parameter and use that to pass player1 and player2

File: bin/main.rb [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    move = gets.chomp
+    while !game.options.include? move.to_i
+      print "\nPleas select from 1 to 9: "
+      move = gets.chomp
+    end
+    while !board.add_map(move.to_i,player1.player.to_s)
+      print "\nOption already used, select a different option: "
+      move = gets.chomp
+    end
+    board.winner = 1 if board.check_map(move.to_i,player1.player.to_s)
```
Comment (Ispirett) [inline bin/main.rb]: This code can be wrapped in a method

File: bin/main.rb [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    move = gets.chomp
+    while !game.options.include? move.to_i
+      print "\nPleas select from 1 to 9: "
+      move = gets.chomp
+    end
+    while !board.add_map(move.to_i,player2.player.to_s)
+      print "\nOption already used, select a different option: "
+      move = gets.chomp
+    end
+    board.winner = 2 if board.check_map(move.to_i,player2.player.to_s)
```
Comment (Ispirett) [inline bin/main.rb]: this code was already created and the code can be reused if it was within a method and add a parameter and use that to pass player1 and player2

**Pull request merged**